### PR TITLE
 ENH: Added compatibility for the NAG Fortran compiler, nagfor

### PR DESCRIPTION
**Pull request by andu97**
This adds the newer NAG Fortran compiler, nagfor, to distutils.

- nag.py modified to include nagfor
- Added tests for the version pattern
- Only for linux atm, as unable to test other OS

### CODE REVIEW

File: numpy/distutils/tests/test_fcompiler_nagfor.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+nagfor_version_strings = [('nagfor', 'NAG Fortran Compiler Release '
+                           '6.2(Chiyoda) Build 6200', '6.2'),
+                          ('nagfor', 'NAG Fortran Compiler Release '
+                           '6.1(Tozai) Build 6136', '6.1'),
+                          ('nagfor', 'NAG Fortran Compiler Release '
+                           '6.0(Hibiya) Build 1021', '6.0'),
+                          ('nagfor', 'NAG Fortran Compiler Release '
+                           '5.3.2(971)', '5.3.2')]
+
+nag_version_strings = [('nag', 'NAGWare Fortran 95 compiler Release 5.1'
```
Comment (charris) [inline numpy/distutils/tests/test_fcompiler_nagfor.py]: That's a heck of a version string. Should there be a ',' at the end of the first line?

Comment (charris) [reply numpy/distutils/tests/test_fcompiler_nagfor.py]: Does `nag_version_strings` get used anywhere?

Comment (andu97) [reply numpy/distutils/tests/test_fcompiler_nagfor.py]: Yep the version string is ridiculous but it's correct. The old compiler outputted it's own release as well as all the ones before it.
Fixed the test now to include all the version strings now, good spot.

**Comment by charris** [comment]
Thanks @andu97 .

**Pull request merged**